File: CartMinder/app/src/main/java/id/ac/ui/cs/mobileprogramming/kevinlh/cartminder/dao/CartDAO.java
package id.ac.ui.cs.mobileprogramming.kevinlh.cartminder.dao;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

import id.ac.ui.cs.mobileprogramming.kevinlh.cartminder.model.Cart;

@Dao
public interface CartDAO {
    @Insert
    long insert(Cart cart);

    @Update
    void update(Cart cart);

    @Delete
    void delete(Cart cart);

    @Query("SELECT * FROM cart_table ORDER BY hour ASC, minute ASC")
    LiveData<List<Cart>> getCartsOrdered();

    @Query("SELECT * FROM cart_table")
    LiveData<List<Cart>> getCarts();
}

File: CartMinder2/app/src/main/java/id/ac/ui/cs/mobileprogramming/kevinlh/cartminder2/database/dao/CartDao.kt
package id.ac.ui.cs.mobileprogramming.kevinlh.cartminder2.database.dao

import androidx.room.*
import id.ac.ui.cs.mobileprogramming.kevinlh.cartminder2.model.Cart
import kotlinx.coroutines.flow.Flow


@Dao
interface CartDao {
    @Insert
    suspend fun insert(cart: Cart): Long

    @Update
    suspend fun update(cart: Cart)

    @Delete
    suspend fun delete(cart: Cart)

    @Query("SELECT * FROM cart_table WHERE id=:cartId")
    suspend fun findById(cartId: Long): Cart

    @Query("SELECT * FROM cart_table")
    fun getCarts(): Flow<List<Cart>>
}